src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -1254,15 +1259,15 @@ private function discoverNamespace(\DOMXPath $domxpath, string $prefix): ?string
             return $this->namespaces[$prefix];
         }
 
-        if (\array_key_exists($prefix, $this->cachedNamespaces)) {
-            return $this->cachedNamespaces[$prefix];
+        if ($this->cachedNamespaces->offsetExists($prefix)) {
+            return $this->cachedNamespaces->offsetGet($prefix);
         }
 
         // ask for one namespace, otherwise we'd get a collection with an item for each node
         $namespaces = $domxpath->query(sprintf('(//namespace::*[name()="%s"])[last()]', $this->defaultNamespacePrefix === $prefix ? '' : $prefix));
 
         $namespace = ($node = $namespaces->item(0)) ? $node->nodeValue : null;
-        $this->cachedNamespaces[$prefix] = $namespace;
+        $this->cachedNamespaces->offsetSet($prefix, $namespace);
```

Comment:
using `this->cachedNamespaces[$prefix] = $namespace;` is fine. that's what `ArrayAccess` is for.

Reply:
I am thankful for your valuable improvement ideas but please let us not discuss every unimportant detail of my implementation. I know I have an ArrayIterator here in opposite to most public ArrayIterator|array|... usages so I decided to directly take its functions.